### PR TITLE
Add checkout feature through stripe

### PR DESCRIPTION
**Pull request by ZeeshanAzhar23**

### CODE REVIEW

File: app/controllers/purchase/checkouts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+class Purchase::CheckoutsController < ApplicationController
+  #before_action :authenticate_user!
+ # before_action :retrieve_price_id, only: [:create]
```
Comment (javier459) [inline app/controllers/purchase/checkouts_controller.rb]: remove commented code.

File: app/controllers/purchase/checkouts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  def create
+      @plan=Plan.find(params[:plan_id])
+      price=@plan.price_id
```
Comment (javier459) [inline app/controllers/purchase/checkouts_controller.rb]: spacing issue

File: app/controllers/purchase/checkouts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      #price = params[:pricing_id]
```
Comment (javier459) [inline app/controllers/purchase/checkouts_controller.rb]: remove commented code

File: app/controllers/purchase/checkouts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        client_reference_id: current_user.id,
+        success_url: root_url + 'success?session_id={CHECKOUT_SESSION_ID}',
+        cancel_url: root_url,
+        payment_method_types: ['card'],
+        mode: 'subscription',
+        # customer_email: current_user.email,
+        line_items: [{
+          quantity: 1,
+          price: price,
+          #price: price,
```
Comment (javier459) [inline app/controllers/purchase/checkouts_controller.rb]: remove commented code.

File: app/controllers/purchase/checkouts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  # def retrieve_price_id
+  #   # price=Stripe::Price.search({
+  #   #   query: 'active:\'true\' AND metadata[\'order_id\']:\'6735\'',
+  #   # }).data[9]
+  #   p=Plan.new
+  #   puts p.return_price_id
+  #   puts "checking price"
+  # end
```
Comment (javier459) [inline app/controllers/purchase/checkouts_controller.rb]: remove commented code

**Review by javier459 — CHANGES_REQUESTED** [review]

**Review by ZeeshanAzhar23 — COMMENTED** [review]
Kindly review the changes

**Review by javier459 — APPROVED** [review]

**Pull request merged**